src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -428,6 +434,68 @@ zocl_xclbin_same_uuid(struct drm_zocl_dev *zdev, xuid_t *uuid)
 	    uuid_equal(uuid, zocl_xclbin_get_uuid(zdev)));
 }
 
+static int
+zocl_load_aie_only_pdi(struct drm_zocl_dev *zdev, struct axlf *axlf,
+                        char __user *xclbin, struct sched_client_ctx *client)
+{
+        uint64_t size;
+        char *pdi_buf = NULL;
+        int ret;
+
+        if (client && client->aie_ctx == ZOCL_CTX_SHARED) {
+                DRM_ERROR("%s Shared context can not load xclbin", __func__);
+                return -EPERM;
+        }
+
+        size = zocl_read_sect(PDI, &pdi_buf, axlf, xclbin);
+        if (size == 0)
+                return 0;
+
+        ret = zocl_fpga_mgr_load(zdev, pdi_buf, size, FPGA_MGR_PARTIAL_RECONFIG);
+        vfree(pdi_buf);
+
+	/* Mark AIE out of reset state after load PDI */
+        if (zdev->aie) {
+                mutex_lock(&zdev->aie_lock);
+                zdev->aie->aie_reset = false;
+                mutex_unlock(&zdev->aie_lock);
+        }
+
+        return ret;
+}
+
+/*
+ * Cache the xclbin blob so that it can be shared by processes.
+ *
+ * Note: currently, we only cache xclbin blob for AIE only xclbin to
+ *       support AIE multi-processes. For AIE only xclbin, we load
+ *       the PDI to AIE even it has been loaded. But if a process is
```

Comment:
So we can load AIE multiple times by calling into zocl_load_aie_pdi and each time we will cache the axlf into zocl->axlf?

It seems each time we will vmalloc a new buffer but do not check if the axlf should be freed? That's a memory leak.